dbt/adapters/default.py

```diff
@@ -0,0 +1,527 @@
+import copy
+import multiprocessing
+import re
+import time
+import yaml
+
+from contextlib import contextmanager
+
+import dbt.exceptions
+import dbt.flags
+
+from dbt.contracts.connection import validate_connection
+from dbt.logger import GLOBAL_LOGGER as logger
+from dbt.schema import Column
+
+
+lock = multiprocessing.Lock()
```

Comment:
yeah, I think that's what we want right now. does that seem right to you?

one other thing here. I thought a little bit about how we could support connections of multiple types in a single run -- if we were to do that, we'd need to change this and the cache to be unique-per-connection-type.